### PR TITLE
Finished Travel-Related Pages and Added Search Page

### PR DESCRIPTION
**Pull request by kimberlypn**
## Changes:
- Added "Search" and "Booked Trips" pages to nav bar
- Added `booked` field to `TravelDates` table. Used to determine whether trip shows up on "Travel Dates" page vs. "Booked Trips" page
- Changed type of `price_limit` in `TravelDates` table from `float` to `integer` since that made more sense
- Designed "Travel Dates" page:
  - Displays all dates that users have set aside for travel (i.e. not yet booked)
  - There is a "Search" button that users can use to make a request to the flights API to see if there are any available flights to that destination within that date range and for that budget (not yet implemented)
  - Users can also edit or delete these records (edit not yet implemented)
- Designed "Booked Trips" page:
  - Displays all trips that users have booked, organized by trips this month and future trips
  - Users can "unbook" a trip (which will delete the record) or edit the trip (edit not yet implemented)
- Designed "Past Trips" page:
  - Displays all trips that users have gone on (i.e. was booked and end date has passed)
  - Users can edit or delete these records (edit not yet implemented)
- Renamed component files to match component names
- Added PropTypes for any component that does not call `dispatch()`
- Added a bunch of trip data to `seeds.exs`
- Removed user's name from nav bar because the text was overflowing. They can view their name on the profile page anyway. 

## TODO:
- Add ability to create a `TravelDate` record
- Add constraint to `TravelDates` table to ensure that `start_date <= end_date`
- Implement search and edit buttons
- Add PropTypes for components that call `dispatch()`
- Might want to create a `BookedTrip` table since a `BookedTrip` has a lot more information (`hotel`, `departureTime`, `arrivalTime`, etc.). Or, we could add these fields to `TravelDate` and set them to `null` until a trip has been booked. The former would require more changes I think but would be more organized.

### CODE REVIEW

File: travelpal/assets/js/components/friend-card.jsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Row, Col } from 'reactstrap';
 import api from '../api';
 
 // Renders the details of an individual friend as a card
-export default function Friend(props) {
+export default function FriendCard(props) {
```
Comment (Guo-William) [inline travelpal/assets/js/components/friend-card.jsx]: I think you should name the file as you would name the component. So in this case the file name should be FriendCard.

Comment (kimberlypn) [reply travelpal/assets/js/components/friend-card.jsx]: You want me to change the file name from `friend-card.jsx` to `FriendCard.jsx`?

Comment (kimberlypn) [reply travelpal/assets/js/components/friend-card.jsx]: I just thought hyphenated lowercase was the name convention for `.jsx` files.

File: travelpal/assets/js/components/travel-dates.jsx [hunk 2]
```diff
@@ -1,22 +1,50 @@
...
+
+  _.map(props.travelDates, function(tt) {
+    // Convert the start and end date strings to Date objects
+    let startDate = new Date(tt.start_date);
+    let endDate = new Date(tt.end_date);
+    // Grab this user's trips
+    if (props.user == tt.user.id) {
+      // Trips that have passed
+      if (endDate < today) {
+
```
Comment (Guo-William) [inline travelpal/assets/js/components/travel-dates.jsx]: Maybe add a TODO line? just as a reminder to fill things out?

File: travelpal/assets/js/components/friend-card.jsx (hunk 1 above)
Comment (Guo-William) [reply travelpal/assets/js/components/friend-card.jsx]: Its just the convention I use but we can keep it as is if you want.

**Comment by Guo-William** [comment]
I see in some files you do Component(props)
I think it would be better to do Component({id, startDate, endDate}) this way we know exactly what a component will need from its argument line. Or add in PropTypes at the bottom of the file.

**Comment by kimberlypn** [comment]
@Guo-William What do you think about the purpose of the "Travel Dates" page (see comment in PR description)?

**Comment by Guo-William** [comment]
So my initial idea for the travel dates page was to allow the user to add dates which they set aside for travel. Also on this page the user could see previous travel dates and their destinations if they set one for it already. Then we can either allow them to go to a page for a select date and show them options there.

**Comment by Guo-William** [comment]
Yes leave the ones with dispatch alone. I think I have a fix for them. If you want to see look at
[https://github.com/Guo-William/Ultimate-Tic-Tac-Toe/blob/master/mathttt/src/containers/GameContainer.js](url)

This was a project I built at my last co-op for ramp up

**Comment by kimberlypn** [comment]
@Guo-William Ok, I'll just leave the dispatch ones alone for now. Link doesn't work btw.

**Comment by Guo-William** [comment]
@kimberlypn  https://github.com/Guo-William/Ultimate-Tic-Tac-Toe My bad

**Comment by kimberlypn** [comment]
Just updated the PR description.

**Comment by mdang8** [comment]
I got a bunch of Brunch errors because of the capitalization when you import components in your JSX files. I renamed them all to match the actual file names and that seemed to fix it.

**Comment by kimberlypn** [comment]
Oh what do you mean? I renamed the imports to match the new file names, which match the component names now.

**Comment by kimberlypn** [comment]
Wait. That's weird. Some of my file name changes didn't save like `nav.jsx`. I'll fix it.

**Comment by kimberlypn** [comment]
It won't let me push the file name changes because it thinks nothing changed. ._.

**Comment by mdang8** [comment]
Hmm I can rename them and push the changes?

**Comment by kimberlypn** [comment]
Nvm, I was able to change it using `git mv <old> <new>`. Thanks, though. (:

**Comment by mdang8** [comment]
I'm assuming the `price_limit` field in `TravelDates` is a float because users can enter cents amounts right?

**Comment by kimberlypn** [comment]
That's what I thought originally, but then it didn't really make sense for a user to say, "I'm going to spend no more than $3500.75" on this trip, you know?

If we add a `BookedTrips` table, then using a float might make more sense because then the user can add the exact cost of the trip after they've paid.

**Comment by mdang8** [comment]
Ohhhh nvm nvm I misread it as you changed it from `integer` to `float` 🤦. Yeah that makes more sense.

**Pull request merged**